server/buildbuddy_server/buildbuddy_server.go

```diff
@@ -248,15 +261,31 @@ func (s *BuildBuddyServer) UpdateGroup(ctx context.Context, req *grpb.UpdateGrou
 	if err := s.authorizeGroupAccess(ctx, req.GetId()); err != nil {
 		return nil, err
 	}
-	group := &tables.Group{
-		GroupID: req.GetId(),
+	var group *tables.Group
+	var err error
+	urlIdentifier := strings.TrimSpace(req.GetUrlIdentifier())
+
+	// Make sure the new URL identifier isn't taken by a different group
+	// than the one being updated.
+	if urlIdentifier != "" {
+		group, err = userDB.GetGroupByURLIdentifier(ctx, urlIdentifier)
+		if err != nil && gstatus.Code(err) != gcodes.NotFound {
+			return nil, err
+		}
+		if group != nil && group.GroupID != req.GetId() {
+			return nil, status.InvalidArgumentError("URL is already in use")
+		}
 	}
-	if err := userDB.FillGroup(ctx, group); err != nil {
-		return nil, err
+
+	if group == nil {
```

Comment:
I'm slightly confused by this logic -- feels like we should be doing:

if req.GetUrlIdent() {
  // read group by url ident
} else if req.GetId() {
  // read group by ID
} else {
  // some error case
}

Reply:
Ah yep, that's much cleaner. Thanks!